core/utility/ast-utils.ts

```diff
@@ -152,26 +181,44 @@ export function addImportToNgModuleMetadata(options: AddSymbolToNgModuleOptions)
 }
 
 export function addImportToModule(options: AddImportToModuleOptions): Rule {
+  return _addImportToFile({
+    filePath: options.modulePath,
+    importName: options.importName,
+    importFrom: options.importFrom
+  });
+}
+
+export function addImportToComponent(options: AddImportToComponentOptions): Rule {
+  return _addImportToFile({
+    filePath: options.componentPath,
+    importName: options.importName,
+    importFrom: options.importFrom
+  });
+}
+
+export function addChangeDetectionToComponent(options: addChangeDetectionToComponentOptions): Rule {
   return (host: Tree) => {
-    if (!options.modulePath) {
+    if (!options.componentPath) {
       return host;
     }
 
-    const text = host.read(options.modulePath);
+    const text = host.read(options.componentPath);
     if (text === null) {
-      throw new SchematicsException(`File ${options.modulePath} does not exist.`);
+      throw new SchematicsException(`File ${options.componentPath} does not exist.`);
     }
 
     const sourceText = text.toString('utf-8');
-    const source = ts.createSourceFile(options.modulePath, sourceText, ts.ScriptTarget.Latest, true);
-
-    const change = insertImport(source, options.modulePath, options.importName, options.importFrom);
+    const source = ts.createSourceFile(options.componentPath, sourceText, ts.ScriptTarget.Latest, true);
+    const insertProperty = ',\n  changeDetection: ChangeDetectionStrategy.OnPush';
 
-    if (change instanceof InsertChange) {
-      const recorder = host.beginUpdate(options.modulePath);
-      recorder.insertLeft(change.pos, change.toAdd);
-      host.commitUpdate(recorder);
+    const classDecorator = findNodes(source, ts.SyntaxKind.Decorator)[0];
```

Comment:
@umbarutiba if `findNodes` will return the empty array or null then here will be the error.

Reply:
const classDecorator = findNodes(source, ts.SyntaxKind.Decorator).find((item) => item.getText().includes('@Component'));
    if (!classDecorator) {
      throw new SchematicsException(`File ${options.componentPath} does not have a decorator.`);
    }